keras2onnx/_builtin.py

```diff
@@ -1879,9 +1924,8 @@ def _prepare_StridedSlice(node, target_opset):
 
         shrink_mask = (shrink_axis_mask >> idx) & 1
         if shrink_mask != 0:
-            shrink_begin = begin_item + _cal_tensor_shape(data_input)[idx] if begin_item < 0 else begin_item
-            new_begin.append(shrink_begin)
-            new_end.append(shrink_begin + 1)
+            new_begin.append(begin_item)
```

Comment:
This supports negative value, so don't need `_cal_tensor_shape` which fails on dynamic shape `[None, None, None, 1]`

Reply:
separate this change to [PR](https://github.com/onnx/keras-onnx/pull/499)